cmd/snap/inhibit.go

```diff
@@ -45,6 +46,19 @@ func waitWhileInhibited(snapName string) error {
 	// and then either unlocks it or changes to HintInhibitedForRefresh (see
 	// gateAutoRefreshHookHandler in hooks.go).
 	// waitInhibitUnlock will return also on HintNotInhibited.
+	usedSnapDekstopIntegration := false
```

Comment:
I think this should be `usedSnapDesktopIntegration` (Desktop typo)

Also, I think this entire if statement with contents should be moved to it's own function, maybe with the name `detectSnapDesktopIntegration`, and then this should just be:

`usedSnapDekstopIntegration = detectSnapDesktopIntegration()`

Because right now `waitWhileInhibited` is getting more complex than it needs to be.

Reply:
I'll do it.

But just a question: as I commented previously, what do you think about removing lines 62 to 69, since that is already being done inside `graphicalSessionFlow()`and `textFlow()`? In fact, that code seems to defeat the purpose of those functions, since it waits until the snap is unlocked to call them...

Reply:
I think it's a great idea if it has no side-effects. My suggestion is to make sure there are unit tests covering this to prove the removal of the lines are fine, otherwise prove it with new unit tests.

It might also be a good candidate for a followup PR.

Reply:
Done.